src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -164,9 +164,14 @@ public void Start<TContext>(IHttpApplication<TContext> application)
                     {
                         var parsedAddress = ServerAddress.FromUrl(address);
 
+                        if (parsedAddress.Scheme.Equals("https", StringComparison.OrdinalIgnoreCase))
+                        {
+                            throw new InvalidOperationException($"HTTPS addresses are not supported. Use {nameof(KestrelServerOptions)}.{nameof(KestrelServerOptions.Listen)}() to configure an HTTPS endpoint.");
```

Comment:
I don't like the phrasing of "HTTPS addresses are not supported..."

I would change this to "HTTPS endpoints may only be configured using {nameof(KestrelServerOptions)}.{nameof(KestrelServerOptions.Listen)}()".